File: tests/fixtures/sample.java
/**
 * Sample Java file for testing parser
 * Contains classes, methods, interfaces, imports
 */

package com.example.auth;

import java.util.List;
import java.util.ArrayList;
import java.util.Optional;
import static java.lang.Math.PI;
import com.example.models.User;
import com.example.database.Database;

/**
 * User authentication service
 * Handles login, logout, and session management
 */
public class AuthService {
    private Database dbClient;
    private static final int SESSION_TIMEOUT = 3600; // 1 hour in seconds
    private static final String API_VERSION = "v1";

    /**
     * Initialize auth service with database client
     */
    public AuthService(Database dbClient) {
        this.dbClient = dbClient;
    }

    /**
     * Authenticate user with credentials
     *
     * @param email User email address
     * @param password User password
     * @return User object if authentication succeeds, empty otherwise
     */
    public Optional<User> login(String email, String password) {
        if (email == null || email.isEmpty() || password == null || password.isEmpty()) {
            throw new IllegalArgumentException("Email and password are required");
        }

        User user = dbClient.query("SELECT * FROM users WHERE email = ?", email);

        if (user == null) {
            return Optional.empty();
        }

        boolean isValid = verifyPassword(password, user.getPasswordHash());

        if (!isValid) {
            return Optional.empty();
        }

        return Optional.of(user);
    }

    /**
     * Verify password against hash
     */
    private boolean verifyPassword(String password, String hash) {
        // Simplified for testing
        return password.equals(hash);
    }

    /**
     * Create new user session
     */
    public String createSession(String userId) {
        String sessionId = generateSessionId();
        long expiresAt = System.currentTimeMillis() + (SESSION_TIMEOUT * 1000L);

        dbClient.query(
            "INSERT INTO sessions (id, user_id, expires_at) VALUES (?, ?, ?)",
            sessionId, userId, expiresAt
        );

        return sessionId;
    }

    /**
     * Generate random session ID
     */
    private String generateSessionId() {
        return java.util.UUID.randomUUID().toString();
    }
}

/**
 * User role enumeration
 */
enum UserRole {
    ADMIN,
    MODERATOR,
    USER
}

/**
 * Permission checker interface
 */
interface PermissionChecker {
    boolean hasPermission(User user, UserRole requiredRole);
}

/**
 * Utility class for permission checks
 */
public class PermissionUtils {
    /**
     * Check user role and permissions
     *
     * @param user User object
     * @param requiredRole Required role for access
     * @return true if user has required permission
     */
    public static boolean hasPermission(User user, UserRole requiredRole) {
        int userLevel = getRoleLevel(user.getRole());
        int requiredLevel = getRoleLevel(requiredRole);

        return userLevel >= requiredLevel;
    }

    /**
     * Get numeric level for role
     */
    private static int getRoleLevel(UserRole role) {
        switch (role) {
            case ADMIN:
                return 3;
            case MODERATOR:
                return 2;
            case USER:
                return 1;
            default:
                return 0;
        }
    }
}

File: tests/fixtures/sample.cpp
/**
 * Sample C++ file for testing parser
 * Contains classes, functions, namespaces, and templates
 */

#include <iostream>
#include <string>
#include <vector>
#include "database.h"

using namespace std;

// Constants
const int SESSION_TIMEOUT = 3600;
const string API_VERSION = "v1";

// User role enumeration
enum UserRole {
    ROLE_ADMIN,
    ROLE_MODERATOR,
    ROLE_USER
};

// User struct
struct User {
    string id;
    string email;
    string passwordHash;
    UserRole role;
};

// Forward declaration
class Database;

/**
 * Authentication service class
 * Handles user login and session management
 */
class AuthService {
private:
    Database* dbClient;
    int sessionTimeout;

    /**
     * Verify password against hash
     */
    bool verifyPassword(const string& password, const string& hash) {
        // Simplified for testing
        return password == hash;
    }

    /**
     * Generate random session ID
     */
    string generateSessionId() {
        return "session_" + to_string(time(nullptr));
    }

public:
    /**
     * Constructor
     */
    AuthService(Database* db) : dbClient(db), sessionTimeout(SESSION_TIMEOUT) {}

    /**
     * Destructor
     */
    ~AuthService() {}

    /**
     * Authenticate user with credentials
     *
     * @param email User email address
     * @param password User password
     * @return Pointer to User object if authentication succeeds, nullptr otherwise
     */
    User* login(const string& email, const string& password) {
        if (email.empty() || password.empty()) {
            throw invalid_argument("Email and password are required");
        }

        User* user = dbClient->query("SELECT * FROM users WHERE email = ?", email);

        if (user == nullptr) {
            return nullptr;
        }

        bool isValid = verifyPassword(password, user->passwordHash);

        if (!isValid) {
            return nullptr;
        }

        return user;
    }

    /**
     * Create new user session
     */
    string createSession(const string& userId) {
        string sessionId = generateSessionId();

        dbClient->execute(
            "INSERT INTO sessions (id, user_id, expires_at) VALUES (?, ?, ?)",
            sessionId, userId, time(nullptr) + sessionTimeout
        );

        return sessionId;
    }
};

/**
 * Permission utility namespace
 */
namespace PermissionUtils {
    /**
     * Check if user has required permission level
     */
    bool hasPermission(const User& user, UserRole requiredRole) {
        return user.role >= requiredRole;
    }

    /**
     * Get role level as integer
     */
    int getRoleLevel(UserRole role) {
        switch (role) {
            case ROLE_ADMIN:
                return 3;
            case ROLE_MODERATOR:
                return 2;
            case ROLE_USER:
                return 1;
            default:
                return 0;
        }
    }
}

/**
 * Template function for calculating complexity
 */
template<typename T>
T calculateMax(const vector<T>& data) {
    if (data.empty()) {
        throw invalid_argument("Data cannot be empty");
    }

    T maxValue = data[0];
    for (const auto& value : data) {
        if (value > maxValue) {
            maxValue = value;
        }
    }

    return maxValue;
}

/**
 * Main function
 */
int main() {
    cout << "Authentication Service v" << API_VERSION << endl;
    return 0;
}
